homeassistant/components/withings/common.py

```diff
@@ -497,23 +550,42 @@ def webhook_config(self) -> WebhookConfig:
     def _do_interval(
         self, action: Callable[[], Coroutine], delay: timedelta, interval: timedelta
     ) -> Callable:
+        """Run action now and schedule it run in an interval for future calls.
+
+        This method has the same functionality of async_track_time_interval except this
+        method also runs the action immediately (with slight delay).
+        :param action: The async function to call.
+        :param delay: Delay the first run by some time. This is good for adding jitter
+        to API calls.
+        :param interval: How much time should elapse between repeats calls of "action".
+        :return: A function for cancelling all future calls.
+        """
         cancel_func1 = None
         cancel_func2 = None
 
         @callback
-        def runner(now: datetime.datetime) -> None:
-            self._hass.async_run_job(action())
+        async def runner(now: datetime.datetime) -> None:
+            await self._hass.async_add_job(action)
 
         def cancel() -> None:
             nonlocal cancel_func1
             nonlocal cancel_func2
-            cancel_func1()
-            cancel_func2()
 
+            if cancel_func1:
```

Comment:
This will never be unset, even if it is no longer attached.

Reply:
I don't understand, `cancel_func1` is set to none a couple lines later.

Reply:
Right, but if func1 had already run, it cannot be canceled anymore and will print a warning. So you need to unset `cancel_func1` in the callback that `func1` is calling (in this case in `runner`)